torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -414,44 +443,21 @@ struct NCCLTraceBuffer {
     result.insert(result.end(), entries_.begin(), entries_.begin() + next_);
     // query any remaining events
     for (auto& r : result) {
-      if (r.start_ != nullptr) {
-        bool started = true;
-        for (auto& ev : *r.start_) {
-          if (!ev.query()) {
-            started = false;
-            break;
-          }
-        }
-        if (started) {
-          r.state_ = "started";
-        }
-        r.start_ = nullptr;
-      }
-      if (r.end_ != nullptr) {
-        bool completed = true;
-        for (auto& ev : *r.end_) {
-          if (!ev.query()) {
-            completed = false;
-            break;
-          }
-        }
-        if (completed) {
-          r.state_ = "completed";
-        }
-        r.end_ = nullptr;
-      }
+      update_state(r);
+      r.start_ = r.end_ = nullptr;
```

Comment:
why clear the events here?  (if its not actually retired, and if we wanted to dump again later, this would be counterproductive?)